### PR TITLE
cmp/contact

### PR DESCRIPTION
**Pull request by Lakshi99**
I integrated the EmailJS into the Contact Us page. It includes the following changes:

- Added EmailJS service to handle form submissions.
- Updated the contact form to use refs for capturing input values.
- Implemented functionality to send an email upon form submission using EmailJS.

### CODE REVIEW

File: skillsync/src/Pages/Contact/Contact.js [hunk 1]
```diff
@@ -8,16 +8,24 @@ import phone from "../../assets/phone.jpg";
...
+    message: messageRef.current.value,
+   };
+
+   emailjs
+      .send(
         "service_vyb0ww4",
         "template_lp1roh3",
-        form.current,
+        templateParams,
         "lXtWrN83VcO2t54Is"
```
Comment (bihanviranga) [inline skillsync/src/Pages/Contact/Contact.js]: Is this an API key? @Lakshi99

Comment (Lakshi99) [reply skillsync/src/Pages/Contact/Contact.js]: No aiya, It is an object that holds the dynamic data(user name, user email and message) that passes to the EmailJS template.

**Comment by bihanviranga** [comment]
I'm asking about the string after templateParams, that goes like lXtWr...?

**Comment by Lakshi99** [comment]
Yes aiya it is the public key.

**Comment by bihanviranga** [comment]
> Yes aiya it is the public key.

In that case, let's move this value to the env file. Keys shouldn't be in the code because it is a security concern and considered a bad practice. Other than that, the PR looks good to me.

**Comment by Lakshi99** [comment]
> > Yes aiya it is the public key.
> 
> In that case, let's move this value to the env file. Keys shouldn't be in the code because it is a security concern and considered a bad practice. Other than that, the PR looks good to me.

Ok aiya. I'll make the changes.

**Comment by RavinduLakshitha** [comment]
great work.

**Review by RavinduLakshitha — APPROVED** [review]
okay. it,s good.

File: skillsync/src/Pages/Contact/Contact.js [hunk 2]
```diff
@@ -6,20 +6,43 @@ import address from "../../assets/address.jpg";
...
+  // console.log('Sending email with params:', templateParams);
+  // console.log('Using API Key:', apiKey);
+
+   emailjs
+   .send("service_vyb0ww4", "template_lp1roh3", templateParams, apiKey)
+      // .send(
+      //   "service_vyb0ww4",
+      //   "template_lp1roh3",
+      //   templateParams,
+      //   // "lXtWrN83VcO2t54Is"
```
Comment (bihanviranga) [inline skillsync/src/Pages/Contact/Contact.js:43]: Hiding the API key is useless if it's still in the code, even as a comment. It should be properly hidden so remove this as well.

File: skillsync/src/Pages/Contact/Contact.js [hunk 3]
```diff
@@ -6,20 +6,43 @@ import address from "../../assets/address.jpg";
...
+    user_name: nameRef.current.value,
+    user_email: emailRef.current.value,
+    message: messageRef.current.value,
+   };
+
+   if (!apiKey) {
+    console.error('EmailJS API key is not defined.');
+    return;
+  }
+  console.log('EmailJS API Key:', apiKey);
```
Comment (bihanviranga) [inline skillsync/src/Pages/Contact/Contact.js:30]: Since the keys are secret/sensitive data, let's not log it either.

**Pull request merged**